lib/vanagon/utilities.rb

```diff
@@ -281,5 +281,16 @@ def erb_file(erbfile, outfile = nil, remove_orig = false, opts = { :binding => b
       FileUtils.rm_rf erbfile if remove_orig
       outfile
     end
+
+    # Final
+    # - 1.0.0
+    # - 3.0.5.1
+    # - 4.3.0-5
+    # Nonfinal
+    # - 1.0.0-34-g1sadjf9283
+    # - 4.2.1.2344.gfdqw83rh
+    def final_version?(version)
+      !version.match(/g[a-f0-9]{7}$/)
```

Comment:
What are the possible formats that you'd expect to see this method parsing? Since you're writing a parser, just laying out as many of the expected formats you think you'll have to parse might provide some insight into how the parser should work.